### ISSUE TITLE
Method names with the suffix _0 generate wrong contract targets

### ISSUE THREAD

**Issue by wadoon** [KeYProject/key#1539]
This issue was created at [git.key-project.org](https://git.key-project.org/key/key/-/issues/1539) where the discussions are preserved.
    
----
    
When loading the following Java file (C.java) into KeY 2.7-1688 ...

```
class C {
  /*@ public normal_behavior
    @ ensures true;
    (at)*/
  void f_0() {}

  /*@ public normal_behavior
    @ ensures true;
    (at)*/
  void f_1() {}

  /*@ public normal_behavior
    @ ensures true;
    (at)*/
  void f0() {}
}
```

... I would expect that three contract targets can be selected in the proof management dialog, namely `C::f_0()`, `C::f_1()`, and `C::f0()`.
However, instead if `C::f_0()`, `C::f()` appears in the dialog:

![image](/uploads/cca8c78f1db5e3463faf779217fb3c0d/image.png)

In other words, KeY does not seem to allow methods with names ending in `_0` (while `0` and `_1` work as suffixes).

When I go on and try to prove a contract which calls `C::f_0()`, I get an exception ...

```
java.lang.NullPointerException
	at de.uka.ilkd.key.rule.UseOperationContractRule.getActualSelf(UseOperationContractRule.java:234)
	at de.uka.ilkd.key.rule.UseOperationContractRule.computeInstantiation(UseOperationContractRule.java:585)
	at de.uka.ilkd.key.rule.UseOperationContractRule.instantiate(UseOperationContractRule.java:449)
	at de.uka.ilkd.key.rule.UseOperationContractRule.isApplicable(UseOperationContractRule.java:611)
	at de.uka.ilkd.key.proof.BuiltInRuleAppIndex.scanSimplificationRule(BuiltInRuleAppIndex.java:135)
	at de.uka.ilkd.key.proof.BuiltInRuleAppIndex.scanAddedFormulas(BuiltInRuleAppIndex.java:184)
	at de.uka.ilkd.key.proof.BuiltInRuleAppIndex.sequentChanged(BuiltInRuleAppIndex.java:173)
	at de.uka.ilkd.key.proof.RuleAppIndex.sequentChanged(RuleAppIndex.java:380)
	at de.uka.ilkd.key.proof.Goal.fireSequentChanged(Goal.java:251)
	at de.uka.ilkd.key.proof.Goal.setSequent(Goal.java:403)
	at de.uka.ilkd.key.rule.executor.javadl.FindTacletExecutor.apply(FindTacletExecutor.java:143)
	at de.uka.ilkd.key.rule.Taclet.apply(Taclet.java:912)
	at de.uka.ilkd.key.rule.TacletApp.execute(TacletApp.java:422)
	at de.uka.ilkd.key.proof.Goal.apply(Goal.java:677)
	at de.uka.ilkd.key.prover.impl.ApplyStrategy.applyAutomaticRule(ApplyStrategy.java:125)
	at de.uka.ilkd.key.prover.impl.ApplyStrategy.doWork(ApplyStrategy.java:145)
	at de.uka.ilkd.key.prover.impl.ApplyStrategy.executeStrategy(ApplyStrategy.java:284)
	at de.uka.ilkd.key.prover.impl.ApplyStrategy.start(ApplyStrategy.java:252)
	at de.uka.ilkd.key.prover.impl.ApplyStrategy.start(ApplyStrategy.java:234)
	at de.uka.ilkd.key.prover.impl.ApplyStrategy.start(ApplyStrategy.java:217)
	at de.uka.ilkd.key.macros.StrategyProofMacro.applyTo(StrategyProofMacro.java:122)
	at de.uka.ilkd.key.macros.AbstractProofMacro.applyTo(AbstractProofMacro.java:89)
	at de.uka.ilkd.key.control.DefaultProofControl$MacroThread.run(DefaultProofControl.java:159)
```

... which, I would guess, happens because the call to `C::f_0()` can not be resolved (because this method has been falsely registered as `C::f()`).

The easy workaround is to not use method names ending in `_0`.
    
----
    
Information: 
* created_at: 2020-07-05T13:15:20.596Z
* updated_at: 2020-07-05T13:15:20.596Z
* closed_at: None  (closed_by: )
* milestone: 
* user_notes_count: 0

**Comment by FliegendeWurst** [comment]
Related #1717

**Comment by FliegendeWurst** [comment]
Duplicate of #1717, fixed in #3231
